### PR TITLE
[#152392124] Secure, full version of new product page

### PR DESCRIPTION
**Pull request by chrisfarms**
## What

Our new product page is a Sinatra app. It was not configured to escape data being output from `.erb` templates, and this led to an XSS vulnerability in our new forms. 

This PR deploys the full new product page (including the contact forms) with security improvements. This contrasts with #31, which deploys a cut-down-to-be-secure version of the new product page.

### Immediate XSS mitigation

* We now use [erubis](https://rubygems.org/gems/erubis/versions/2.7.0) for rendering `.erb` templates. 
* We have configured erubis/sinatra to automatically escape all data output using `<%= user_provided_data %>`. Data can be output without escaping using `<%== safe_or_already_escaped_data %>`. 
* We have vetted all occurrences of `<%`. The only uses of `<%==` are where data has already been escaped (for instance, outputting a rendered partial template.)

See the 1st commit for more information.

### Application hardening

We have hardened the Sinatra application, to mitigate future vulnerabilities and reduce their likelihood by ensuring safe configuration:
  
* A `Content-Security-Policy` header [[1](https://developer.mozilla.org/en-US/docs/Web/HTTP/Headers/Content-Security-Policy), [2](https://content-security-policy.com)] is now sent in app responses. See the 2nd commit for more information.
* We have added tests against @samcrang's XSS exploit and for the presence of the `Content-Security-Policy` header. See the 3th commit for more information.
* We have not configured CSRF protection, but we have explicitly pointed it out for if we need it in future. See the 4rd commit for more information.

## How to review

1. Run the new tests with `make test` and ensure they fail.
2.. Get the app running in your dev environment. Verify the `Content-Security-Policy` header is present and that Sam's original attack (visit `/contact-us?person_name="><script>alert(%27%27)</script>`) no longer works.
3. Consider whether you're confident in this app being redeployed.

This will probably be merged *after* https://github.com/alphagov/paas-product-page/pull/31, which removes lots of dynamic behaviour in order to make the product page safe to redeploy. It includes many of the same security-related commits as this PR.

## Who can review

Not @46bit @chrisfarms @carolinegreen @samcrang.

### CODE REVIEW

**Comment by 46bit** [comment]
I've reviewed all occurrences of `<%` and this LGTM. We're reviewing other possible security issues.

File: app.rb [hunk 1]
```diff
@@ -6,12 +6,15 @@
...
 require './models/deskpro'
 require './models/forms'
+require 'erubis'
 
 # App is the main Sinatra application
 class App < Sinatra::Base
 
 	set :sprockets, Sprockets::Environment.new
 	set :title, 'GOV.UK Platform as a Service'
+	set :erb, :escape_html => true
```
Comment (46bit) [inline app.rb]: This enables the `erubis` gem's automatic escaping.

File: app.rb [hunk 2]
```diff
@@ -6,12 +6,15 @@
...
+	use Rack::Protection::ContentSecurityPolicy, :default_src => 'none', :script_src => '\'self\' www.google-analytics.com', :style_src => '\'self\'', :img_src => '\'self\' www.google-analytics.com', :connect_src => '\'self\'', :frame_src => '\'self\'', :font_src => '\'self\' data:', :object_src => '\'self\'', :media_src => '\'self\''
```
Comment (46bit) [inline app.rb]: This configures the `Content-Security-Policy` header.

File: assets/js/application.js [hunk 3]
```diff
@@ -1,2 +1,4 @@
+//= require google-analytics.js
+//= require jquery-3.2.1.slim.min.js
```
Comment (46bit) [inline assets/js/application.js]: Our new `Content-Security-Policy` forbids inline `<script>`s from running. We therefore embed the GA code in the main compiled JavaScript file.

**Comment by 46bit** [comment]
**This was not merged. Don't worry.** We merged #31, which is these security-related commits *plus a commit to remove lots of potentially-unsafe dynamic content.* We presume that Github saw we'd merged all the commits in this PR and automatically marked it as merged.

**Comment by dcarley** [comment]
> This was not merged. Don't worry.

I was wondering, thanks 😄

**Pull request merged**